### PR TITLE
[RLlib] Make coingame a tests again

### PR DESCRIPTION
**Pull request by ArturNiederfahrenhorst**
## Why are these changes needed?

This PR includes the coingame env in tests again and fixes issues with it.

## Related issue number

Fixes https://discuss.ray.io/t/marl-mapping-policy-examples-not-working/9081
and https://github.com/ray-project/ray/issues/31853

### CODE REVIEW

File: rllib/examples/env/coin_game_non_vectorized_env.py [hunk 1]
```diff
@@ -46,14 +52,13 @@ def __init__(self, config: Optional[Dict] = None):
...
         self.n_features = self.grid_size**2 * (2 * self.NUM_AGENTS)
-        self.OBSERVATION_SPACE = gym.spaces.Box(
+        self.observation_space = gym.spaces.Box(
             low=0, high=1, shape=(self.grid_size, self.grid_size, 4), dtype="uint8"
         )
 
         self.step_count_in_current_episode = None
         if self.output_additional_info:
             self._init_info()
-        self.seed(seed=config.get("seed", None))
```
Comment (ArturNiederfahrenhorst) [inline rllib/examples/env/coin_game_non_vectorized_env.py:56]: Neither essential to this example, does self.seed exist anymore.

Comment (avnishn) [reply rllib/examples/env/coin_game_non_vectorized_env.py:56]: ok so to be clear, this is the broken code and we deleted it, but can users set the seed via a call to reset now?

**Review by avnishn — COMMENTED** [review]
qq below :)

File: rllib/examples/env/coin_game_non_vectorized_env.py (hunk 1 above)
Comment (ArturNiederfahrenhorst) [reply rllib/examples/env/coin_game_non_vectorized_env.py:56]: Yes, the reset method calls from gymnasium.utils.seeding with the user provided seed!

Comment (avnishn) [reply rllib/examples/env/coin_game_non_vectorized_env.py:56]: mk :)

**Review by avnishn — APPROVED** [review]

**Review by gjoliver — APPROVED** [review]

**Pull request merged**